Publishing for linux-bionix-x64 using NativeAOT fails

### Description

Publishing a .net class library with the linux-bionic-x64 rid using NativeAOT fails with a clang error. Publishing with linux-bionic-arm64 works fine. The following is the command output with both targets;
```
D:\natTest>dotnet publish . -r linux-bionic-arm64 -p:DisableUnsupportedError=true -p:PublishAotUsingRuntimePack=true
MSBuild version 17.8.3+195e7f5a3 for .NET
  Determining projects to restore...
  Restored D:\natTest\natTest.csproj (in 360 ms).
  natTest -> D:\natTest\bin\Release\net8.0\linux-bionic-arm64\natTest.dll
  Generating native code
  natTest -> D:\natTest\bin\Release\net8.0\linux-bionic-arm64\publish\

D:\natTest>dotnet publish . -r linux-bionic-x64 -p:DisableUnsupportedError=true -p:PublishAotUsingRuntimePack=true
MSBuild version 17.8.3+195e7f5a3 for .NET
  Determining projects to restore...
  Restored D:\natTest\natTest.csproj (in 304 ms).
  natTest -> D:\natTest\bin\Release\net8.0\linux-bionic-x64\natTest.dll
  Generating native code
lld : error : unknown argument: --version-script=obj\Release\net8.0\linux-bionic-x64\native\natTest.exports [D:\natTest\natTest.csproj]
lld : error : unknown argument: -rpath [D:\natTest\natTest.csproj]
lld : error : unknown argument: --build-id=sha1 [D:\natTest\natTest.csproj]
lld : error : unknown argument: -z [D:\natTest\natTest.csproj]
lld : error : unknown argument: -z [D:\natTest\natTest.csproj]
lld : error : unknown argument: --eh-frame-hdr [D:\natTest\natTest.csproj]
lld : error : unknown argument: --discard-all [D:\natTest\natTest.csproj]
lld : error : unknown argument: -T [D:\natTest\natTest.csproj]
clang : error : linker command failed with exit code 1 (use -v to see invocation) [D:\natTest\natTest.csproj]
C:\nuget\packages\microsoft.dotnet.ilcompiler\8.0.0\build\Microsoft.NETCore.Native.targets(364,5): error MSB3073: The command ""clang" "obj\Release\net8.0\linux-bionic-x
64\native\natTest.o" -o "bin\Release\net8.0\linux-bionic-x64\native\natTest.so" -Wl,--version-script=obj\Release\net8.0\linux-bionic-x64\native\natTest.exports -gz=zlib
-fuse-ld=lld C:\nuget\packages\microsoft.netcore.app.runtime.nativeaot.linux-bionic-x64\8.0.0/runtimes/linux-bionic-x64/\native\libbootstrapperdll.o C:\nuget\packages\mi
crosoft.netcore.app.runtime.nativeaot.linux-bionic-x64\8.0.0/runtimes/linux-bionic-x64/\native\libRuntime.WorkstationGC.a C:\nuget\packages\microsoft.netcore.app.runtime
.nativeaot.linux-bionic-x64\8.0.0/runtimes/linux-bionic-x64/\native\libeventpipe-disabled.a C:\nuget\packages\microsoft.netcore.app.runtime.nativeaot.linux-bionic-x64\8.
0.0/runtimes/linux-bionic-x64/\native\libstdc++compat.a C:\nuget\packages\microsoft.netcore.app.runtime.nativeaot.linux-bionic-x64\8.0.0/runtimes/linux-bionic-x64/\nativ
e\libSystem.Native.a C:\nuget\packages\microsoft.netcore.app.runtime.nativeaot.linux-bionic-x64\8.0.0/runtimes/linux-bionic-x64/\native\libSystem.IO.Compression.Native.a
 C:\nuget\packages\microsoft.netcore.app.runtime.nativeaot.linux-bionic-x64\8.0.0/runtimes/linux-bionic-x64/\native\libSystem.Security.Cryptography.Native.OpenSsl.a -g -
Wl,-rpath,"$ORIGIN" -Wl,--build-id=sha1 -Wl,--as-needed -Wl,-e0x0 -pthread -ldl -lz -llog -lm -shared -Wl,-z,relro -Wl,-z,now -Wl,--eh-frame-hdr -Wl,--discard-all -Wl,--
gc-sections -Wl,-T,"obj\Release\net8.0\linux-bionic-x64\native\sections.ld"" exited with code 1. [D:\natTest\natTest.csproj]
```

A reproduction can be found here, https://github.com/emmauss/natTest .

### Reproduction Steps

1. Create a Class Library project
2. Set the following project properties;
```
<NativeLib>Shared</NativeLib>
<InvariantGlobalization>true</InvariantGlobalization>
<PublishAot>true</PublishAot>
```
3. Set the top of the PATH environmental variable to the android ndk: `set path=C:\Android\android-sdk\ndk\25.1.8937393\toolchains\llvm\prebuilt\windows-x86_64\bin;%path%`
4. Publish with linux-bionic-x64 target: `dotnet publish . -r linux-bionic-x64 -p:DisableUnsupportedError=true -p:PublishAotUsingRuntimePack=true`

### Expected behavior

Publishing with linux-bionic-x64 should succeed like linux-bionic-arm64 does.

### Actual behavior

clang fails with this error;
```
lld : error : unknown argument: -rpath [D:\natTest\natTest.csproj]
lld : error : unknown argument: --build-id=sha1 [D:\natTest\natTest.csproj]
lld : error : unknown argument: -z [D:\natTest\natTest.csproj]
lld : error : unknown argument: -z [D:\natTest\natTest.csproj]
lld : error : unknown argument: --eh-frame-hdr [D:\natTest\natTest.csproj]
lld : error : unknown argument: --discard-all [D:\natTest\natTest.csproj]
lld : error : unknown argument: -T [D:\natTest\natTest.csproj]
clang : error : linker command failed with exit code 1 (use -v to see invocation) [D:\natTest\natTest.csproj]
C:\nuget\packages\microsoft.dotnet.ilcompiler\8.0.0\build\Microsoft.NETCore.Native.targets(364,5): error MSB3073: The command ""clang" "obj\Release\net8.0\linux-bionic-x
64\native\natTest.o" -o "bin\Release\net8.0\linux-bionic-x64\native\natTest.so" -Wl,--version-script=obj\Release\net8.0\linux-bionic-x64\native\natTest.exports -gz=zlib
-fuse-ld=lld C:\nuget\packages\microsoft.netcore.app.runtime.nativeaot.linux-bionic-x64\8.0.0/runtimes/linux-bionic-x64/\native\libbootstrapperdll.o C:\nuget\packages\mi
crosoft.netcore.app.runtime.nativeaot.linux-bionic-x64\8.0.0/runtimes/linux-bionic-x64/\native\libRuntime.WorkstationGC.a C:\nuget\packages\microsoft.netcore.app.runtime
.nativeaot.linux-bionic-x64\8.0.0/runtimes/linux-bionic-x64/\native\libeventpipe-disabled.a C:\nuget\packages\microsoft.netcore.app.runtime.nativeaot.linux-bionic-x64\8.
0.0/runtimes/linux-bionic-x64/\native\libstdc++compat.a C:\nuget\packages\microsoft.netcore.app.runtime.nativeaot.linux-bionic-x64\8.0.0/runtimes/linux-bionic-x64/\nativ
e\libSystem.Native.a C:\nuget\packages\microsoft.netcore.app.runtime.nativeaot.linux-bionic-x64\8.0.0/runtimes/linux-bionic-x64/\native\libSystem.IO.Compression.Native.a
 C:\nuget\packages\microsoft.netcore.app.runtime.nativeaot.linux-bionic-x64\8.0.0/runtimes/linux-bionic-x64/\native\libSystem.Security.Cryptography.Native.OpenSsl.a -g -
Wl,-rpath,"$ORIGIN" -Wl,--build-id=sha1 -Wl,--as-needed -Wl,-e0x0 -pthread -ldl -lz -llog -lm -shared -Wl,-z,relro -Wl,-z,now -Wl,--eh-frame-hdr -Wl,--discard-all -Wl,--
gc-sections -Wl,-T,"obj\Release\net8.0\linux-bionic-x64\native\sections.ld"" exited with code 1. [D:\natTest\natTest.csproj]
```

### Regression?

_No response_

### Known Workarounds

_No response_

### Configuration

OS: Windows 11 x86-64
.Net Version: 8.0.0 Release

### Other information

_No response_

--- a/src/coreclr/nativeaot/BuildIntegration/Microsoft.DotNet.ILCompiler.SingleEntry.targets
+++ b/src/coreclr/nativeaot/BuildIntegration/Microsoft.DotNet.ILCompiler.SingleEntry.targets
@@ -4,10 +4,10 @@
     <!-- Define the name of the runtime specific compiler package to import -->
     <_hostOS>$(NETCoreSdkPortableRuntimeIdentifier.SubString(0, $(NETCoreSdkPortableRuntimeIdentifier.LastIndexOf('-'))))</_hostOS>
 
-    <_targetOS>$(RuntimeIdentifier.SubString(0, $(RuntimeIdentifier.LastIndexOf('-'))))</_targetOS>
-    <_indexOfPeriod>$(_targetOS.IndexOf('.'))</_indexOfPeriod>
-    <_targetOS Condition="'$(_indexOfPeriod)' &gt; -1">$(_targetOS.SubString(0, $(_indexOfPeriod)))</_targetOS>
-    <_targetOS Condition="$(_targetOS.StartsWith('win'))">win</_targetOS>
+    <_originalTargetOS>$(RuntimeIdentifier.SubString(0, $(RuntimeIdentifier.LastIndexOf('-'))))</_originalTargetOS>
+    <_indexOfPeriod>$(_originalTargetOS.IndexOf('.'))</_indexOfPeriod>
+    <_originalTargetOS Condition="'$(_indexOfPeriod)' &gt; -1">$(_originalTargetOS.SubString(0, $(_indexOfPeriod)))</_originalTargetOS>
+    <_originalTargetOS Condition="$(_originalTargetOS.StartsWith('win'))">win</_originalTargetOS>
 
     <!-- On non-Windows, determine _hostArchitecture from NETCoreSdkPortableRuntimeIdentifier -->
     <_hostArchitecture Condition="'$(OS)' != 'Windows_NT'">$(NETCoreSdkPortableRuntimeIdentifier.SubString($([MSBuild]::Add($(NETCoreSdkPortableRuntimeIdentifier.LastIndexOf('-')), 1))))</_hostArchitecture>
@@ -18,11 +18,13 @@
     <_targetArchitecture>$(RuntimeIdentifier.SubString($([MSBuild]::Add($(RuntimeIdentifier.LastIndexOf('-')), 1))))</_targetArchitecture>
 
     <_hostPackageName>runtime.$(_hostOS)-$(_hostArchitecture).Microsoft.DotNet.ILCompiler</_hostPackageName>
-    <_targetPackageName>runtime.$(_targetOS)-$(_targetArchitecture).Microsoft.DotNet.ILCompiler</_targetPackageName>
+    <_targetPackageName>runtime.$(_originalTargetOS)-$(_targetArchitecture).Microsoft.DotNet.ILCompiler</_targetPackageName>
 
-    <!-- From this point onwards, we will treat linux-musl as linux -->
-    <_linuxLibcFlavor Condition="$(_targetOS.StartsWith('linux-'))">$(_targetOS.SubString(6))</_linuxLibcFlavor>
-    <_targetOS Condition="$(_targetOS.StartsWith('linux'))">linux</_targetOS>
+    <!-- Treat linux-musl and linux-bionic etc. as linux -->
+    <_targetOS>$(_originalTargetOS)</_targetOS>
+    <_linuxToken>linux-</_linuxToken>
+    <_linuxLibcFlavor Condition="$(_targetOS.StartsWith($(_linuxToken)))">$(_targetOS.SubString($(_linuxToken.Length)))</_linuxLibcFlavor>
+    <_targetOS Condition="$(_targetOS.StartsWith($(_linuxToken)))">linux</_targetOS>
   </PropertyGroup>
 
   <PropertyGroup>

--- a/src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.targets
+++ b/src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.targets
@@ -33,7 +33,7 @@ The .NET Foundation licenses this file to you under the MIT license.
       <FullRuntimeName Condition="'$(ServerGarbageCollection)' == 'true'">libRuntime.ServerGC</FullRuntimeName>
 
       <CrossCompileRid />
-      <CrossCompileRid Condition="!$(RuntimeIdentifier.StartsWith('$(_hostOS)-')) or !$(RuntimeIdentifier.EndsWith('-$(_hostArchitecture)'))">$(RuntimeIdentifier)</CrossCompileRid>
+      <CrossCompileRid Condition="$(_hostOS) != $(_originalTargetOS) or !$(RuntimeIdentifier.EndsWith('-$(_hostArchitecture)'))">$(RuntimeIdentifier)</CrossCompileRid>
 
       <CrossCompileArch />
       <CrossCompileArch Condition="$(CrossCompileRid.EndsWith('-x64'))">x86_64</CrossCompileArch>

--- a/src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.targets
+++ b/src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.targets
@@ -33,8 +33,7 @@ The .NET Foundation licenses this file to you under the MIT license.
       <FullRuntimeName Condition="'$(ServerGarbageCollection)' == 'true'">libRuntime.ServerGC</FullRuntimeName>
 
       <CrossCompileRid />
-      <CrossCompileRid Condition="!$(RuntimeIdentifier.EndsWith('-$(_hostArchitecture)'))">$(RuntimeIdentifier)</CrossCompileRid>
-      <CrossCompileRid Condition="'$(CrossCompileRid)' == '' and '$(_linuxLibcFlavor)' == 'bionic'">$(RuntimeIdentifier)</CrossCompileRid>
+      <CrossCompileRid Condition="!$(RuntimeIdentifier.StartsWith('$(_hostOS)-')) or !$(RuntimeIdentifier.EndsWith('-$(_hostArchitecture)'))">$(RuntimeIdentifier)</CrossCompileRid>
 
       <CrossCompileArch />
       <CrossCompileArch Condition="$(CrossCompileRid.EndsWith('-x64'))">x86_64</CrossCompileArch>

